File: modules/EnchantmentsEnhance-Plugin/src/main/java/org/pixeltime/enchantmentsenhance/gui/Clickable.java
package org.pixeltime.enchantmentsenhance.gui;

import org.bukkit.inventory.ItemStack;
import org.pixeltime.enchantmentsenhance.manager.CompatibilityManager;

public abstract class Clickable {
    public abstract ItemStack getItem(String player);

    public ItemStack getGlowingItem(String playerName) {
        return CompatibilityManager.glow.addGlow(getItem(playerName));
    }

    public abstract int getPosition();

}

File: modules/EnchantmentsEnhance-Plugin/src/main/kotlin/org/pixeltime/enchantmentsenhance/gui/Clickable.kt
package org.pixeltime.enchantmentsenhance.gui

import org.bukkit.inventory.ItemStack
import org.pixeltime.enchantmentsenhance.manager.CompatibilityManager

abstract class Clickable {

    abstract val position: Int
    abstract fun getItem(player: String): ItemStack

    open fun getGlowingItem(playerName: String): ItemStack {
        return CompatibilityManager.glow.addGlow(getItem(playerName))
    }

}
